### PR TITLE
Emitting 'InProgress' ETW event for short functions.

### PR DESCRIPTION
**Pull request by akurmi**

### CODE REVIEW

**Comment by mathewc** [comment]
Please provide a detailed description for this - its not clear at all from looking or reading your title what the actual issue is.

File: test/WebJobs.Script.Tests/MetricsEventManagerTests.cs [hunk 1]
```diff
@@ -72,20 +72,6 @@ public class MetricsEventManagerTests
                 string.Format("There are events with different execution id. List:{0} Invalid entries:{1}",
                     SerializeFunctionExecutionEventArguments(argsList),
                     SerializeFunctionExecutionEventArguments(invalidArgsList)));
-
```
Comment (mathewc) [inline test/WebJobs.Script.Tests/MetricsEventManagerTests.cs]: Why removing all this? It doesn't seem that your changes should have affected this test?

Comment (akurmi) [inline test/WebJobs.Script.Tests/MetricsEventManagerTests.cs]: This change would require tweaks in the validation logic as the number and types of ETW events have changed. I didn't want to spend much time tweaking this validation routine, I would introduce this back in a separate commit after making it work with this change.

Comment (mathewc) [inline test/WebJobs.Script.Tests/MetricsEventManagerTests.cs]: Why not just update the test to validate the new expected behavior?

Comment (akurmi) [inline test/WebJobs.Script.Tests/MetricsEventManagerTests.cs]: I thought we were in hurry to release this tomorrow :). If that is not the case then I would fix this tomorrow morning.

Comment (mathewc) [inline test/WebJobs.Script.Tests/MetricsEventManagerTests.cs]: We'll have time - just fix it in the morning and we'll get it in :)

**Pull request closed without merging**